packages/frontend-core/src/fetch/ViewV2Fetch.js

```diff
@@ -53,7 +54,10 @@ export default class ViewV2Fetch extends DataFetch {
       this.options.sortColumn = definition.sort.field
       this.options.sortOrder = definition.sort.order
     }
-    if (!filter?.length && definition.query?.length) {
+
+    const parsed = utils.processSearchFilters(filter)
+
+    if (!parsed?.groups?.length && definition.query?.groups?.length) {
```

Comment:
I see that this is comparing against `definition.query.groups`, but for all views created before this change, their definitions will still contain the old flat structure since there's no migration. Is that something we need to account for? Or maybe those saved filters will have gone though `processSearchFilters` by the time they get there.

Reply:
Yep, `definition.query` is processed as part of the `getInitialData` process in the parent class `DataFetch`. The processed value is then placed in the store and used here.